[CALCITE-3733] MySQL JDBC rewrite should rewrite CAST as TIMESTAMP to CAST as DATETIME (Vineet Garg)

vineetgarg02 opened the pull request.

DonnyZone commented:
I noticed that the timestamp type is supported in MySQL (https://dev.mysql.com/doc/refman/8.0/en/datetime.html).
Is the `cast (x as timestamp)` not supported?

vineetgarg02 commented:
> I noticed that the timestamp type is supported in MySQL (https://dev.mysql.com/doc/refman/8.0/en/datetime.html).
> Is the `cast (x as timestamp)` not supported?

@DonnyZone Yes `cast(x as timestamp)` is not supported in MySQL

danny0405 commented on core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:
```diff
@@ -1834,6 +1834,18 @@ private static String toSql(RelNode root, SqlDialect dialect) {
+  @Test void testMySqlCastToTimestamp() {
+    //final String query = "select cast(\"hire_date\" as TIMESTAMP), \"employee_id\" "
+     //   + "from \"employee\" ";
+    //final String expected = "select cast(\"hire_date\" as DATE_TIME), \"employee_id\" "
```
Remove the comment

danny0405 commented:
Thanks for the fix @vineetgarg02 , let's give this patch a better commit message:
"In JDBC adapter, when generating SQL for MYSQL, generate TIMESTAMP type as DATETIME for CAST"

vineetgarg02 commented:
> Thanks for the fix @vineetgarg02 , let's give this patch a better commit message:
> "In JDBC adapter, when generating SQL for MYSQL, generate TIMESTAMP type as DATETIME for CAST"

@danny0405  Thanks for taking a look, i have addressed your comments/concerns.
